api/pkg/service/catalog/catalog.go

```diff
@@ -139,11 +140,18 @@ func (s *service) CatalogError(ctx context.Context, p *catalog.CatalogErrorPaylo
 		}
 	}
 
+	var errorTypes []string
```

Comment:
also we can remove the switch case, it seems we won't be having error type other than the three as we explicitly add them
```
rs[item.Type] = append(rs[item.Type], item.Detail)
```
this would work

Reply:
Yep this should work